### PR TITLE
Feature/conf matrix double scored data

### PR DESCRIPTION
**Pull request by mulhod**
Add in h1/h2 confusion matrix in reports.

~~Still a draft for the moment.~~

You can try it out on the `rsmtool` examples.

### CODE REVIEW

File: rsmtool/analyzer.py [hunk 1]
```diff
@@ -1803,6 +1796,23 @@ def run_prediction_analyses(self, data_container, configuration):
...
+                    df_preds_second_score_double_scored[f"sc{human}"]
+                )
+            human1_scores = df_preds_second_score_double_scored["sc1_round"].astype("int64")
+            human2_scores = df_preds_second_score_double_scored["sc2_round"].astype("int64")
+            conf_matrix_h1h2 = confusion_matrix(human1_scores, human2_scores)
+            labels = sorted(pd.concat([human1_scores, human2_scores]).unique())
+            df_confmatrix_h1h2 = pd.DataFrame(
+                conf_matrix_h1h2, index=labels, columns=labels
+            ).transpose()
+
```
Comment (desilinguist) [inline rsmtool/analyzer.py:1815]:
```suggestion
        # compute a human1-human2 confusion matrix, if possible
        if include_second_score:
            df_preds_second_score_double_scored = df_preds_second_score.copy().dropna(
                subset=["sc1", "sc2"]
            )
            human1_scores = df_preds_second_score_double_scored["sc1"].round().astype("int64")
            human2_scores = df_preds_second_score_double_scored["sc2"].round().astype("int64")
            conf_matrix_h1h2 = confusion_matrix(human1_scores, human2_scores)
            labels = sorted(pd.concat([human1_scores, human2_scores]).unique())
            df_confmatrix_h1h2 = pd.DataFrame(
                conf_matrix_h1h2, index=labels, columns=labels
            ).transpose()

```

File: rsmtool/analyzer.py [hunk 2]
```diff
@@ -1803,6 +1796,23 @@ def run_prediction_analyses(self, data_container, configuration):
...
 
+        # compute a human1-human2 confusion matrix, if possible
+        if include_second_score:
+            df_preds_second_score_double_scored = df_preds_second_score.copy().dropna(
+                subset=["sc1", "sc2"]
+            )
+            for human in [1, 2]:
+                df_preds_second_score_double_scored[f"sc{human}_round"] = np.round(
+                    df_preds_second_score_double_scored[f"sc{human}"]
+                )
```
Comment (desilinguist) [inline rsmtool/analyzer.py:1807]:
```suggestion
```

File: rsmtool/analyzer.py [hunk 3]
```diff
@@ -1803,6 +1796,23 @@ def run_prediction_analyses(self, data_container, configuration):
...
+            human1_scores = df_preds_second_score_double_scored["sc1_round"].astype("int64")
+            human2_scores = df_preds_second_score_double_scored["sc2_round"].astype("int64")
```
Comment (desilinguist) [inline rsmtool/analyzer.py:1809]:
```suggestion
            human1_scores = df_preds_second_score_double_scored["sc1"].round().astype("int64")
            human2_scores = df_preds_second_score_double_scored["sc2"].round().astype("int64")
```

**Review by desilinguist — CHANGES_REQUESTED** [review]
Looks great! Just a couple of minor suggestions ...

**Comment by desilinguist** [comment]
@mulhod you will also need to:
- update the [documentation](https://rsmtool.readthedocs.io/en/v11.0.1/usage_rsmtool.html#human-human-consistency) to include the new output file in the intermediate outputs. 
- update the intermediate files notebook to link to the new file.
-

**Comment by mulhod** [comment]
> @mulhod you will also need to:
> 
> * update the [documentation](https://rsmtool.readthedocs.io/en/v11.0.1/usage_rsmtool.html#human-human-consistency) to include the new output file in the intermediate outputs.
> * update the intermediate files notebook to link to the new file.

I have addressed point #1. #2 is already addressed, I think (I didn't have to do anything special for this, I believe):
Screenshot 2023-08-24 at 11 36 38 PM

**Review by desilinguist — APPROVED** [review]
LGTM!

**Comment by desilinguist** [comment]
@tamarl08 do you mind taking a look as well?

**Review by tazin-afrin — APPROVED** [review]
Looks good to me too!

**Comment by desilinguist** [comment]
Yes, I missed that! I see it twice too.

**Comment by mulhod** [comment]
@tamarl08 Good catch! I see what's happening. Minor issue.

**Review by tamarl08 — APPROVED** [review]

**Pull request merged**